numpy/lib/function_base.py

```diff
@@ -764,6 +764,19 @@ def copy(a, order='K', subok=False):
     >>> x[0] == z[0]
     False
 
+    Note that, NumPy will not copy the elements of object arrays:
+
+    >>> a = np.array([1, 2, 3])
+    >>> b = np.copy(np.array([a, [a, a]], dtype=object))
+
+    Any updates on b changes a:
+
+    >>> b[0][0] = 10
+    >>> np.alltrue(b[0] == a)
+    True
+    >>> np.alltrue(b[1][0] == a)
+    True
```

Comment:
I think it would be better to use a simpler element to explain object arrays

```suggestion
    Note that ``np.copy`` will not copy the elements of object arrays:
  
    >>> obj = dict()
    >>> a = np.array([obj], dtype=object)
    >>> b = np.copy(a)
    >>> b[0] is a[0]
    True

    To copy the elements as well, use `copy.deepcopy`:

    >>> c = copy.deepcopy(a)
    >>> c[0] is a[0]
    False
    
    Remember that "ragged arrays" are nothing but object arrays with arrays as their elements, so they behave in the same way.
```